tinygrad/tensor.py

```diff
@@ -808,15 +808,15 @@ def calc_dim(tensor_dim:int) -> int:
         ret = ret.permute(*range(first_dim, first_dim+len(big_shape)), *range(0, first_dim), *range(first_dim+len(big_shape), ret.ndim))
     return ret
 
-  def __setitem__(self, indices, v:Union[Tensor, ConstType]):
-    if isinstance(self.device, str) and self.device.startswith("DISK"): return self.__getitem__(indices).assign(v)
+  def __setitem__(self, indices, v:Union[Tensor, ConstType]) -> None:
+    if isinstance(self.device, str) and self.device.startswith("DISK"): self.__getitem__(indices).assign(v)
     # TODO: broadcast v to the shape here, refactor for const v and one way broadcast_shape
     if not isinstance(v, Tensor): v = Tensor(v, self.device, self.dtype)
     assign_to = self.realize().__getitem__(indices)
     # NOTE: we check that indices is valid first
-    assert self.lazydata.contiguous(), "setitem target needs to be contiguous"
+    assert cast(LazyBuffer, self.lazydata).st.contiguous, "setitem target needs to be contiguous"
```

Comment:
how does this work for multi? should it be `all(lb.st.contiguous for lb in self.lbs)`?

Reply:
yeah that makes sense.

the other way is to do `self.lazydata = self.lazydata.contiguous()` to force contiguous inside setitem.
positive: don't need to `.contiguous()` in external implementation before setitem
negative: still need to `.contiguous()` before when using TinyJIT (for reasons I'm still trying to figure out) so there's a difference in usage...